File: task320/src/main/java/com/example/task320/controller/NewsController.java
package com.example.task320.controller;

import com.example.task320.dto.request.NewsRequestTo;
import com.example.task320.dto.response.NewsResponseTo;
import com.example.task320.service.NewsService;
import jakarta.validation.Valid;
import org.springframework.http.*;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1.0/news")
public class NewsController {

    private final NewsService service;

    public NewsController(NewsService service) {
        this.service = service;
    }

    @PostMapping
    public ResponseEntity<NewsResponseTo> create(@Valid @RequestBody NewsRequestTo body) {
        return ResponseEntity.status(HttpStatus.CREATED).body(service.create(body));
    }

    @GetMapping
    public List<NewsResponseTo> getAll() {
        return service.getAll();
    }

    @GetMapping("/{id}")
    public NewsResponseTo getById(@PathVariable long id) {
        return service.getById(id);
    }

    @PutMapping("/{id}")
    public NewsResponseTo update(@PathVariable long id, @Valid @RequestBody NewsRequestTo body) {
        return service.update(id, body);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable long id) {
        service.delete(id);
    }
}

File: 351004/Leshok/Lab6/controllers/NewsController.js
// controllers/NewsController.js
const newsService = require('../services/NewsService');

exports.getAll = async (req, res, next) => {
  try {
    const all = await newsService.getAll();
    res.json(all);
  } catch (err) { next(err); }
};

exports.getById = async (req, res, next) => {
  try {
    const item = await newsService.getById(req.params.id);
    res.json(item);
  } catch (err) { next(err); }
};

exports.create = async (req, res, next) => {
  try {
    const created = await newsService.create(req.body);
    res.status(201).json(created);
  } catch (err) { next(err); }
};

exports.update = async (req, res, next) => {
  try {
    const updated = await newsService.update(req.params.id, req.body);
    res.json(updated);
  } catch (err) { next(err); }
};

exports.delete = async (req, res, next) => {
  try {
    await newsService.delete(req.params.id);
    res.status(204).send();
  } catch (err) { next(err); }
};